File: java/pgcon1/Q1.java
package pgcon1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Q1 {

    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	String line1 = br.readLine();
    	String line2 = br.readLine();
    	String line3 = br.readLine();
    	int n = Integer.parseInt(line1);
    	String[] cols2 = line2.split(" ");
    	String[] cols3 = line3.split(" ");
    	List<Integer> list2 = new ArrayList<Integer>();
    	List<Integer> list3 = new ArrayList<Integer>();
    	for (int i=0; i<cols2.length; i++) {
    		int v = Integer.parseInt(cols2[i]);
    		if (i == 0 && v == 0) break;
    		if (v > n) continue;
    		list2.add(v);
    	}
    	for (int i=0; i<cols3.length; i++) {
    		int v = Integer.parseInt(cols3[i]);
    		if (i == 0 && v == 0) break;
    		if (v > n) continue;
    		list3.add(v);
    	}
    	List<Integer> result = new ArrayList<Integer>();
    	for (int i=0; i<list3.size(); i++) {
    		int v = list3.get(i);
    		if (!list2.contains(v)) {
    			result.add(v);
    		}
    	}
    	Collections.sort(result);
    	if (result.size() == 0) {
    		System.out.println("None");
    	} else {
	    	for (int i=0; i<result.size(); i++) {
	    		if (i != 0) {
	    			System.out.print(" ");
	    		}
	    		System.out.print(result.get(i));
	    	}
	    	System.out.println("");
    	}
    }

}

File: c/pgcon1/q1.c
//プログラムコンテスト
//C言語用
//ファイル名 q1.c
//作成者 
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LEN ( 10000 )

char line1[MAX_LEN];
char line2[MAX_LEN];
char line3[MAX_LEN];
char *cols2[MAX_LEN];
char *cols3[MAX_LEN];
char list2[MAX_LEN];
char list3[MAX_LEN];
char result[MAX_LEN];

int split(char *str, const char *delim, char *outlist[]) {
    char *token;
    int count = 0;

    token = strtok(str, delim);
    while (token != NULL && count < MAX_LEN) {
        outlist[count++] = token;
        token = strtok(NULL, delim);
    }
    return count;
}

int main() {
    int i, count, n, v, first;

    fgets(line1, sizeof(line1), stdin);
    fgets(line2, sizeof(line2), stdin);
    fgets(line3, sizeof(line3), stdin);
    n = atoi(line1);
    count = split(line2, " ", cols2);
    for (i = 0; i < count; i++) {
        v = atoi(cols2[i]);
        if (i == 0 && v == 0) break;
        if (v > n) continue;
        list2[v] = 1;
    }
    count = split(line3, " ", cols3);
    for (i = 0; i < count; i++) {
        v = atoi(cols3[i]);
        if (i == 0 && v == 0) break;
        if (v > n) continue;
        list3[v] = 1;
    }
    count = 0;
    for (i = 0; i < MAX_LEN; i++) {
        if (list3[i]) {
            if (!list2[i]) {
                result[i] = 1;
                count++;
            }
        }
    }
    if (count == 0) {
        printf("None\n");
    } else {
        first = 1;
        for (i = 0; i < MAX_LEN; i++) {
            if (result[i]) {
                if (!first) printf(" ");
                printf("%d", i);
                first = 0;
            }
        }
    }
    return 0;
}
